### ISSUE TITLE
go-appimage treats executables as libraries to set RPATH

### ISSUE THREAD

**Issue by brunvonlope** [probonopd/go-appimage#314]
Running objdump -c on binaries from an .appimage produced by go-appimage reveals that the tool uses executables paths as suitables to be appended on RPATH, not only the paths of libraries.

As far I know, it is not common dynamically linkining to executables on Linux. That was a thing on very early versions of Windows but not anymore. This seems a error. 

@Samueru-sama probably can share some words too

**Comment by Samueru-sama** [comment]
Yeah that isn't right and quite a problem for gimp since there is a directory for each plugin.

**Comment by probonopd** [comment]
The intended behavior is that only directories containing libraries should be added to RPATH.

**Comment by brunvonlope** [comment]
> can you spot where?

@probonopd GIMP plug-ins are stored in lib path, this way:

`AppDir/usr/lib/arch_triplet/gimp/3.0/plug-ins/plugin_name/plugin_name_executable`

> The intended behavior is that only directories containing libraries should be added to RPATH

What is a library for go_appimage? go-appimage seems to consider a library whatever ELF files that it found recursively on lib path, independently of the *so extension, creating this monstruosity:

![388291568-b0223f31-255c-400f-860a-8a9f697b04d7](https://github.com/user-attachments/assets/77d90c36-b2c1-4229-b60f-7c0c46be20eb)

**Comment by probonopd** [comment]
Does the build in #316 improve the situation?
